src/Datadog.Trace.ClrProfiler.Native/clr_helpers.cpp

```diff
@@ -682,4 +683,304 @@ bool TryParseSignatureTypes(const ComPtr<IMetaDataImport2>& metadata_import,
+                                            const ComPtr<IMetaDataEmit2>& metadata_emit,
+                                            const ComPtr<IMetaDataAssemblyEmit>& assembly_emit,
+                                            mdToken* ret_type_token) {
+  const auto cor_element_type = CorElementType(*p_sig);
+  WSTRING managed_type_name = ""_W;
+
+  switch (cor_element_type) {
+    case ELEMENT_TYPE_VALUETYPE: {
+      ULONG result;
+      result = CorSigUncompressToken(p_sig + 1, ret_type_token);
+      if (result == -1) {
+        Warn("[trace::ReturnTypeTokenforElementType] ELEMENT_TYPE_VALUETYPE failed to find uncompress TypeRef or TypeDef");
+        return false;
+      }
+
+      return true;
+    }
+
+    case ELEMENT_TYPE_VOID:     // 0x01  // System.Void (struct)
+      managed_type_name = "System.Void"_W;
+      break;
+    case ELEMENT_TYPE_BOOLEAN:  // 0x02  // System.Boolean (struct)
+      managed_type_name = "System.Boolean"_W;
+      break;
+    case ELEMENT_TYPE_CHAR:     // 0x03  // System.Char (struct)
+      managed_type_name = "System.Char"_W;
+      break;
+    case ELEMENT_TYPE_I1:       // 0x04  // System.SByte (struct)
+      managed_type_name = "System.SByte"_W;
+      break;
+    case ELEMENT_TYPE_U1:       // 0x05  // System.Byte (struct)
+      managed_type_name = "System.Byte"_W;
+      break;
+    case ELEMENT_TYPE_I2:       // 0x06  // System.Int16 (struct)
+      managed_type_name = "System.Int16"_W;
+      break;
+    case ELEMENT_TYPE_U2:       // 0x07  // System.UInt16 (struct)
+      managed_type_name = "System.UInt16"_W;
+      break;
+    case ELEMENT_TYPE_I4:       // 0x08  // System.Int32 (struct)
+      managed_type_name = "System.Int32"_W;
+      break;
+    case ELEMENT_TYPE_U4:       // 0x09  // System.UInt32 (struct)
+      managed_type_name = "System.UInt32"_W;
+      break;
+    case ELEMENT_TYPE_I8:       // 0x0a  // System.Int64 (struct)
+      managed_type_name = "System.Int64"_W;
+      break;
+    case ELEMENT_TYPE_U8:       // 0x0b  // System.UInt64 (struct)
+      managed_type_name = "System.UInt64"_W;
+      break;
+    case ELEMENT_TYPE_R4:       // 0x0c  // System.Single (struct)
+      managed_type_name = "System.Single"_W;
+      break;
+    case ELEMENT_TYPE_R8:       // 0x0d  // System.Double (struct)
+      managed_type_name = "System.Double"_W;
+      break;
+    case ELEMENT_TYPE_TYPEDBYREF:  // 0X16  // System.TypedReference (struct)
+      managed_type_name = "System.TypedReference"_W;
+      break;
+    case ELEMENT_TYPE_I:           // 0x18  // System.IntPtr (struct)
+      managed_type_name = "System.IntPtr"_W;
+      break;
+    case ELEMENT_TYPE_U:           // 0x19  // System.UIntPtr (struct)
+      managed_type_name = "System.UIntPtr"_W;
+      break;
+    default:
+      return false;
+  }
+
+  // Create reference to Mscorlib
+  mdModuleRef mscorlib_ref;
+  if (!CreateAssemblyRefToMscorlib(assembly_emit, &mscorlib_ref)) {
```

Comment:
If for some reason we didn't return the right primitive type from our integration (`int` in `ExecuteNonQuery`) we need to create a TypeRef to that primitive and return it as the argument for the `unbox.any` instruction.